bfapi/service/users.py

```diff
@@ -173,14 +173,23 @@ def _fetch_geoaxis_profile(access_token: str):
     profile = response.json()
 
     user_id = profile.get('DN')
+    uid = profile.get('UID')
+    firstname = profile.get('firstname')
+    lastname = profile.get('lastname')
     if not user_id:
-        log.error('Geoaxis response missing `DN`: %s', response.text)
-        raise InvalidGeoaxisResponse('missing `DN`', response.text)
+        if lastname and firstname and uid:
+            user_id = 'O=Beachfront,OU=ID.me,CN={}.{},UID={}'.format(lastname, firstname, uid)
+        else:
+            log.error('Geoaxis response missing important values: %s', response.text)
+            raise InvalidGeoaxisResponse('missing important values', response.text)
```

Comment:
Consider adding a debug log for when user_id is populated. Try to always log every possible condition for easier troubleshooting.